### PR TITLE
en: add backup&restore to gcs using br

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed, added or deleted? (Required)

- Add 2 docs:
  - back up data to gcs using br
  - restore data from gcs using br
- Modify 2 docs

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-tidb-operator/pull/326
- Other reference link(s):

### CODE REVIEW

File: en/backup-to-gcs-using-br.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+# Back up Data to GCS Using BR
+
+This document describes how to back up the data of a TiDB cluster in Kubernetes to [Google Cloud Storage](https://cloud.google.com/storage/docs/) (GCS). "Backup" in this document refers to full backup (ad-hoc full backup and scheduled full backup). [BR](https://pingcap.com/docs/stable/br/backup-and-restore-tool/) is used to get the logic backup of the TiDB cluster, and then this backup data is sent to GCS.
+
+The backup method described in this document is implemented using Custom Resource Definition (CRD) in TiDB Operator v1.1 or later versions.
+
+## Ad-hoc full backup
+
+Ad-hoc full backup describes the backup by creating a `Backup` Custom Resource (CR) object. TiDB Operator performs the specific backup operation based on this `Backup` object. If an error occurs during the backup process, TiDB Operator does not retry, and you need to handle this error manually.
```
Comment (shuijing198799) [inline en/backup-to-gcs-using-br.md:14]: `you need` may be change to `customer need` ?

**Comment by shuijing198799** [comment]
rest LGTM

File: en/backup-to-gcs-using-br.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    ```shell
+    kubectl get bk -n test1 -owide
+    ```
+
+<details>
+
+<summary>More descriptions of fields in the <code>Backup</code> CR</summary>
+
+* `.spec.metadata.namespace`: The namespace where the `Backup` CR is located.
+* `.spec.from.host`: The address of the TiDB cluster to be backed up.
```
Comment (TomShawn) [inline en/backup-to-gcs-using-br.md:155]: ```为需要导出的 TiDB 的 service name，例如 `basic-tidb`。``` is missing.

File: en/backup-to-gcs-using-br.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+    ```shell
+    kubectl get bk -n test1 -owide
+    ```
+
+<details>
+
+<summary>More descriptions of fields in the <code>Backup</code> CR</summary>
+
+* `.spec.metadata.namespace`: The namespace where the `Backup` CR is located.
```
Comment (TomShawn) [inline en/backup-to-gcs-using-br.md:154]: Some content is missing in your translation:

* `.spec.tikvGCLifeTime`：备份中的临时 `tikv_gc_lifetime` 时间设置，默认为 72h。

    在备份开始之前，若 TiDB 集群的 `tikv_gc_lifetime` 小于用户设置的 `spec.tikvGCLifeTime`，为了保证备份的数据不被 TiKV GC 掉，TiDB Operator 会在备份前[调节 `tikv_gc_lifetime`](https://docs.pingcap.com/zh/tidb/stable/dumpling-overview#导出大规模数据时的-tidb-gc-设置) 为 `spec.tikvGCLifeTime`。

    备份结束后不论成功或者失败，只要老的 `tikv_gc_lifetime` 比设置的 `.spec.tikvGCLifeTime` 小，TiDB Operator 都会尝试恢复 `tikv_gc_lifetime` 为备份前的值。在极端情况下，TiDB Operator 访问数据库失败会导致 TiDB Operator 无法自动恢复 `tikv_gc_lifetime` 并认为备份失败。

    此时，可以通过下述语句查看当前 TiDB 集群的 `tikv_gc_lifetime`：

    ```sql
    select VARIABLE_NAME, VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME like "tikv_gc_life_time";
    ```

    如果发现 `tikv_gc_lifetime` 值过大（通常为 10m），则需要按照[调节 `tikv_gc_lifetime`](https://docs.pingcap.com/zh/tidb/stable/dumpling-overview#导出大规模数据时的-tidb-gc-设置) 将 `tikv_gc_lifetime` 调回原样。

* `.spec.cleanData`：设置为 true 时删除该 Backup CR 时会同时清除该 CR 备份出的数据，默认为 false。值得注意的是，在 v1.1.2 以及之前版本不存在该字段，且默认在删除 CR 的同时删除备份的文件。若 v1.1.3 及之后版本的用户希望保持该行为，需要设置该字段为 true。

File: en/backup-to-gcs-using-br.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
+---
+title: Back up Data to GCS Using BR
+summary: Learn how to back up data to Google Cloud Storage (GCS) using BR.
+---
+
+# Back up Data to GCS Using BR
+
+This document describes how to back up the data of a TiDB cluster in Kubernetes to [Google Cloud Storage](https://cloud.google.com/storage/docs/) (GCS). "Backup" in this document refers to full backup (ad-hoc full backup and scheduled full backup). [BR](https://pingcap.com/docs/stable/br/backup-and-restore-tool/) is used to get the logic backup of the TiDB cluster, and then this backup data is sent to GCS.
```
Comment (DanielZhangQD) [inline en/backup-to-gcs-using-br.md:8]:
```suggestion
This document describes how to back up the data of a TiDB cluster in Kubernetes to [Google Cloud Storage](https://cloud.google.com/storage/docs/) (GCS). "Backup" in this document refers to full backup (ad-hoc full backup and scheduled full backup). [BR](https://pingcap.com/docs/stable/br/backup-and-restore-tool/) is used to get the backup of the TiDB cluster, and then the backup data is sent to GCS.
```

File: en/backup-to-gcs-using-br.md (hunk 1 above)
Comment (DanielZhangQD) [reply en/backup-to-gcs-using-br.md:14]: I think the current expression is OK, the users are reading this doc.

File: en/restore-from-gcs-using-br.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
+---
+title: Restore Data from GCS Using BR
+summary: Learn how to restore data from Google Cloud Storage (GCS) using BR.
+---
+
+# Restore Data from GCS Using BR
+
+This document describes how to restore the TiDB cluster data backed up using TiDB Operator in Kubernetes. [BR](https://docs.pingcap.com/tidb/dev/backup-and-restore-tool) is used to perform the restoration.
```
Comment (TomShawn) [inline en/restore-from-gcs-using-br.md:8]:
```suggestion
This document describes how to restore the TiDB cluster data backed up using TiDB Operator in Kubernetes. [BR](https://docs.pingcap.com/tidb/dev/backup-and-restore-tool) is used to perform the restore.
```

File: en/restore-from-gcs-using-br.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+The restoration method described in this document is implemented based on Custom Resource Definition (CRD) in TiDB Operator v1.1 or later versions.
```
Comment (TomShawn) [inline en/restore-from-gcs-using-br.md:10]:
```suggestion
The restore method described in this document is implemented based on Custom Resource Definition (CRD) in TiDB Operator v1.1 or later versions.
```

File: en/restore-from-gcs-using-br.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        projectId: ${project-id}
+        secretName: gcs-secret
+        bucket: ${bucket}
+        prefix: ${prefix}
+        # location: us-east1
+        # storageClass: STANDARD_IA
+        # objectAcl: private
+    ```
+
+2. After creating the `Restore` CR, execute the following command to check the restoration status:
```
Comment (TomShawn) [inline en/restore-from-gcs-using-br.md:87]:
```suggestion
2. After creating the `Restore` CR, execute the following command to check the restore status:
```

File: en/backup-to-gcs-using-br.md [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+
+The backup method described in this document is implemented using Custom Resource Definition (CRD) in TiDB Operator v1.1 or later versions.
+
+## Ad-hoc full backup
+
+Ad-hoc full backup describes the backup by creating a `Backup` Custom Resource (CR) object. TiDB Operator performs the specific backup operation based on this `Backup` object. If an error occurs during the backup process, TiDB Operator does not retry, and you need to handle this error manually.
+
+Currently, the above three authorization methods are supported for the ad-hoc full backup. This document provides examples in which the data of the `demo1` TiDB cluster in the `test1` Kubernetes namespace is backed up to AWS storage and all the above methods are used in the examples.
```
Comment (DanielZhangQD) [inline en/backup-to-gcs-using-br.md:16]: This sentence does not match with the Chinese version.

File: en/restore-from-gcs-using-br.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+    ```shell
+    kubectl get rt -n test2 -owide
+    ```
+
+This example restores the backup data stored in the `spec.gcs.prefix` folder of the `spec.gcs.bucket` bucket on GCS to the TiDB cluster `spec.to.host`. For more information on the configuration items of BR and GCS, refer to [`backup-gcs.yaml`](backup-to-gcs-using-br.md#process-of-ad-hoc-full-backup).
+
+<details>
+
+<summary>More descriptions of fields in the <code>Restore<code> CR</summary>
```
Comment (TomShawn) [inline en/restore-from-gcs-using-br.md:99]:
```suggestion
<summary>More descriptions of fields in the <code>Restore</code> CR</summary>
```

File: en/restore-from-gcs-using-br.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+* `.spec.metadata.namespace`: The namespace where the `Restore` CR is located.
+* `.spec.to.host`: The address of the TiDB cluster to be restored.
+* `.spec.to.port`: The port of the TiDB cluster to be restored.
+* `.spec.to.user`: The accessing user of the TiDB cluster to be restored.
+* `.spec.to.tidbSecretName`: The secret of the user password of the `.spec.to.user` TiDB cluster.
```
Comment (TomShawn) [inline en/restore-from-gcs-using-br.md:105]:
```suggestion
* `.spec.to.tidbSecretName`: The secret of the user password of the `.spec.to.user` TiDB cluster to be restored.
```

File: en/restore-from-gcs-using-br.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+* `.spec.to.tlsClientSecretName`: The secret of the certificate used during the restore.
+
+    If TLS is enabled for the TiDB cluster, but you do not want to restore data using the `${cluster_name}-cluster-client-secret` as described in [Enable TLS between TiDB Components](enable-tls-between-components.md), you can use the `.spec.to.tlsClientSecretName` parameter to specify a secret for the restore. To generate the secret, run the following command:
```
Comment (TomShawn) [inline en/restore-from-gcs-using-br.md:108]:
```suggestion
    If [TLS](enable-tls-between-components.md) is enabled for the TiDB cluster, but you do not want to restore data using the `${cluster_name}-cluster-client-secret` as described in [Enable TLS between TiDB Components](enable-tls-between-components.md), you can use the `.spec.to.tlsClientSecretName` parameter to specify a secret for the restore. To generate the secret, run the following command:
```

File: en/backup-to-gcs-using-br.md [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+
+    In the example above, some parameters in `spec.br` can be ignored, such as `logLevel`, `statusAddr`, `concurrency`, `rateLimit`, `checksum`, and `sendCredToTikv`.
+
+    <details>
+    <summary>More description of parameters</summary>
+
+    - `spec.br.cluster`: The name of the cluster to be backed up.
+    - `spec.br.clusterNamespace`: The `namespace` of the cluster to be backed up.
+    - `spec.br.logLevel`: The log level (`info` by default).
+    - `spec.br.statusAddr`: The HTTP port used for listening a process status in BR to facilitate debugging. If not specified, BR does not listen the process status by default.
```
Comment (DanielZhangQD) [inline en/backup-to-gcs-using-br.md:101]:
```suggestion
    - `spec.br.statusAddr`: The listening address through which BR provides statistics. If not specified, BR does not listen on any status address by default.
```

Comment (DanielZhangQD) [reply en/backup-to-gcs-using-br.md:101]: From https://docs.pingcap.com/tidb/stable/backup-and-restore-tool#common-options

File: en/backup-to-gcs-using-br.md [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+    CS supports the following ACL strategies:
+
+    * `authenticatedRead`
+    * `bucketOwnerFullControl`
+    * `bucketOwnerRead`
+    * `private`
+    * `projectPrivate`
+    * `publicRead`
+
+    If you do not configure `storageClass`, the default is `COLDLINE`. See [GCS Documentation](https://cloud.google.com/storage/docs/storage-classes) for details.
```
Comment (DanielZhangQD) [inline en/backup-to-gcs-using-br.md:139]: This needs to update.

File: en/backup-to-gcs-using-br.md [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+<details>
+
+<summary>More descriptions of fields in the <code>Backup</code> CR</summary>
+
+* `.spec.metadata.namespace`: The namespace where the `Backup` CR is located.
+* `.spec.from.host`: The address of the TiDB cluster to be backed up.
+* `.spec.from.port`: The port of the TiDB cluster to be backed up.
+* `.spec.from.user`: The accessing user of the TiDB cluster to be backed up.
+* `.spec.gcs.bucket`: The name of the bucket which stores data.
+* `.spec.gcs.prefix`: This field is used to make up the path of the remote storage: `s3://${.spec.gcs.bucket}/${.spec.gcs.prefix}/backupName`. This field can be ignored.
```
Comment (DanielZhangQD) [inline en/backup-to-gcs-using-br.md:159]: Should be `gcs://`, @shuijing198799 please confirm the format.

File: en/backup-to-gcs-using-br.md [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+* `.spec.from.tidbSecretName`: The secret of the user password of the `.spec.from.user` TiDB cluster.
```
Comment (DanielZhangQD) [inline en/backup-to-gcs-using-br.md:160]:
```suggestion
* `.spec.from.tidbSecretName`: The secret containg the password of the `.spec.from.user` in the TiDB cluster.
```

File: en/restore-from-gcs-using-br.md (hunk 10 above)
Comment (DanielZhangQD) [inline en/restore-from-gcs-using-br.md:105]:
```suggestion
* `.spec.to.tidbSecretName`: The secret containg the password of the `.spec.to.user` in the TiDB cluster.
```

File: en/backup-to-gcs-using-br.md (hunk 14 above)
Comment (shuijing198799) [reply en/backup-to-gcs-using-br.md:159]: yes here should be `gcs://`

**Comment by ran-huang** [comment]
@DanielZhangQD @TomShawn @shuijing198799 comments addressed. PTAL

File: en/backup-to-gcs-using-br.md [hunk 16]
```diff
@@ -0,0 +1,287 @@
...
+    ```
+
+<details>
+
+<summary>More descriptions of fields in the <code>Backup</code> CR</summary>
+
+* `.spec.metadata.namespace`: The namespace where the `Backup` CR is located.
+* `.spec.tikvGCLifeTime`: The temporary `tikv_gc_lifetime` time setting during the backup. Defaults to 72h.
+
+    Before the backup begins, if the `tikv_gc_lifetime` in the TiDB cluster is smaller than the `spec.tikvGCLifeTime` set by the user, TiDB Operator adjusts `tikv_gc_lifetime` to `spec.tikvGCLifeTime`. This operation makes sure that the backup data is not garbage collected by TiKV.
```
Comment (ran-huang) [inline en/backup-to-gcs-using-br.md:157]: The related link (dumpling-overview.md) is not available in English yet.
I'll add those link when this PR (https://github.com/pingcap/docs-cn/pull/3800) is translated.

Comment (TomShawn) [inline en/backup-to-gcs-using-br.md:157]:
```suggestion
    Before the backup begins, if the `tikv_gc_lifetime` setting in the TiDB cluster is smaller than `spec.tikvGCLifeTime` set by the user, TiDB Operator adjusts the value of `tikv_gc_lifetime` to the value of `spec.tikvGCLifeTime`. This operation makes sure that the backup data is not garbage-collected by TiKV.
```

File: en/backup-to-gcs-using-br.md [hunk 17]
```diff
@@ -0,0 +1,287 @@
...
+
+    After the backup, whether the backup is successful or not, if the previous `tikv_gc_lifetime` is smaller than `.spec.tikvGCLifeTime`, TiDB Operator will try to set `tikv_gc_lifetime` to the previous value.
```
Comment (TomShawn) [inline en/backup-to-gcs-using-br.md:159]:
```suggestion
    After the backup, no matter whether the backup is successful or not, as long as the previous `tikv_gc_lifetime` is smaller than `.spec.tikvGCLifeTime`, TiDB Operator will try to set `tikv_gc_lifetime` to the previous value.
```

File: en/backup-to-gcs-using-br.md [hunk 18]
```diff
@@ -0,0 +1,287 @@
...
+
+    In extreme cases, if TiDB Operator fails to access the database, TiDB Operator cannot automatically recover the value of `tikv_gc_lifetime` and thinks the backup fails. At this time, you can view `tikv_gc_lifetime` of the current TiDB cluster using the following statement:
```
Comment (TomShawn) [inline en/backup-to-gcs-using-br.md:161]:
```suggestion
    In extreme cases, if TiDB Operator fails to access the database, TiDB Operator cannot automatically recover the value of `tikv_gc_lifetime` and treats the backup as failed. At this time, you can view `tikv_gc_lifetime` of the current TiDB cluster using the following statement:
```

File: en/backup-to-gcs-using-br.md [hunk 19]
```diff
@@ -0,0 +1,287 @@
...
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    select VARIABLE_NAME, VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME like "tikv_gc_life_time";
+    ```
+
+    If the value of `tikv_gc_lifetime` is set too large (usually 10m), you need to set it back to the previous value.
+
+* `.spec.cleanPolicy`: The backup file clean policy when the backup CR is deleted after the backup is completed.
```
Comment (TomShawn) [inline en/backup-to-gcs-using-br.md:171]:
```suggestion
* `.spec.cleanPolicy`: The clean policy of the backup file when the backup CR is deleted after the backup is completed.
```

File: en/backup-to-gcs-using-br.md [hunk 20]
```diff
@@ -0,0 +1,287 @@
...
+
+    If the value of `tikv_gc_lifetime` is set too large (usually 10m), you need to set it back to the previous value.
+
+* `.spec.cleanPolicy`: The clean policy of the backup file when the backup CR is deleted after the backup is completed.
+
+    Three clean policies are supported:
+
+    * `Retain`: On any circumstances, retain the backup file when deleting the backup CR.
+    * `Delete`: On any circumstances, delete the backup file when deleting the backup CR.
+    * `OnFailure`: If the backup fails, delete the backup file when deleting the backup CR.
```
Comment (DanielZhangQD) [inline en/backup-to-gcs-using-br.md:177]: s/file/data

File: en/backup-to-gcs-using-br.md (hunk 16 above)
Comment (DanielZhangQD) [reply en/backup-to-gcs-using-br.md]: I think we can copy the command here, and no need to link to that doc.

File: en/backup-to-gcs-using-br.md [hunk 21]
```diff
@@ -0,0 +1,299 @@
...
+    select VARIABLE_NAME, VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME like "tikv_gc_life_time";
+    ```
+
+    If the value of `tikv_gc_lifetime` is set too large (usually 10m), you need to set it back to the previous value:
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    update mysql.tidb set VARIABLE_VALUE = '10m' where VARIABLE_NAME = 'tikv_gc_life_time';
+    ```
```
Comment (ran-huang) [inline en/backup-to-gcs-using-br.md:181]: @DanielZhangQD PTAL if these commands are correct.

File: en/backup-to-gcs-using-br.md [hunk 22]
```diff
@@ -0,0 +1,299 @@
...
+
+    Three clean policies are supported:
+
+    * `Retain`: On any circumstances, retain the backup data when deleting the backup CR.
+    * `Delete`: On any circumstances, delete the backup data when deleting the backup CR.
+    * `OnFailure`: If the backup fails, delete the backup data when deleting the backup CR.
+
+    If this field is not configured, or if you configure a value other than the three policies above, the backup data is retained.
+
+    Note that in v1.1.2 and earlier versions, this field does not exist. The backup data is deleted along with the CR by default. For v1.1.3 or later versions, if you want to keep this behavior, set this field to `Delete`.
```
Comment (ran-huang) [inline en/backup-to-gcs-using-br.md:193]: `file` -> `data` updated.

File: en/backup-to-gcs-using-br.md [hunk 23]
```diff
@@ -154,7 +154,13 @@ This document provides examples in which the data of the `demo1` TiDB cluster in
...
 * `.spec.tikvGCLifeTime`: The temporary `tikv_gc_lifetime` time setting during the backup. Defaults to 72h.
 
-    Before the backup begins, if the `tikv_gc_lifetime` setting in the TiDB cluster is smaller than `spec.tikvGCLifeTime` set by the user, TiDB Operator adjusts the value of `tikv_gc_lifetime` to the value of `spec.tikvGCLifeTime`. This operation makes sure that the backup data is not garbage-collected by TiKV.
+    Before the backup begins, if the `tikv_gc_lifetime` setting in the TiDB cluster is smaller than `spec.tikvGCLifeTime` set by the user, TiDB Operator adjusts the value of `tikv_gc_lifetime` to the value of `spec.tikvGCLifeTime`. This operation makes sure that the backup data is not garbage-collected by TiKV:
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    update mysql.tidb set VARIABLE_VALUE = '720h' where VARIABLE_NAME = 'tikv_gc_life_time';
+    ```
```
Comment (DanielZhangQD) [inline en/backup-to-gcs-using-br.md:163]: No need to add command here as this will be done automatically by TiDB Operator.

**Pull request merged**